packages/ui/__stories__/PasswordField.stories.tsx

```diff
@@ -0,0 +1,145 @@
+      value={value}
+      placeholder="Enter the name"
+      label="Wisest jedi"
+      {...eventHandlers}
+      onChange={({ target: { value } }) => setValue(value)}
+    />
+  )
+}
+Default.parameters = {
+  design: {
+    type: 'figma',
+    url: 'https://www.figma.com/file/8mVm6LTbp2Z0RaWWjTZoft/%F0%9F%90%9DHIVE---Hazelcast-Design-System?node-id=479%3A273',
+  },
+}
+
+export const Error = () => (
+  <PasswordField name="name" value="password" placeholder="Enter the name" label="Wisest jedi" error="Dark side" {...eventHandlers} />
+)
+
+export const Hovered = () => (
+  <PasswordField
+    name="name"
+    value="password"
+    placeholder="Enter the name"
+    label="Wisest jedi"
+    className={styles.hover}
+    {...eventHandlers}
+  />
+)
+
+export const Focused = () => (
+  <PasswordField
+    name="name"
+    value="password"
+    placeholder="Enter the name"
+    label="Wisest jedi"
+    className={styles.focus}
+    {...eventHandlers}
+  />
+)
+
+export const FocusedWithError = () => (
+  <PasswordField
+    name="name"
+    value="password"
+    placeholder="Enter the name"
+    label="Wisest jedi"
+    className={styles.focus}
+    error="Dark side"
+    {...eventHandlers}
+  />
+)
+
+export const Disabled = () => (
+  <PasswordField name="name" value="password" placeholder="Enter the name" label="Wisest jedi" disabled {...eventHandlers} />
+)
+
+export const Empty = () => <PasswordField name="name" placeholder="Enter the name" label="Wisest jedi" {...eventHandlers} />
+
+export const Visible = () => {
+  const visibleRef = useRef<VisibleRef>(null)
+
+  useLayoutEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    visibleRef.current!.setVisible(true)
+  }, [])
+
+  return (
+    <PasswordField
+      name="name"
+      value="password"
+      placeholder="Enter the name"
+      label="Wisest jedi"
+      visibleRef={visibleRef}
+      {...eventHandlers}
+    />
+  )
+}
+
+export const WithHelperText = () => (
+  <PasswordField
+    name="name"
+    value="password"
+    label="Wisest jedi"
+    placeholder="Enter the name"
+    helperText="Lorem Ipsum is simply dummy text of the printing and typesetting industry."
+    {...eventHandlers}
+  />
+)
+
+export const PasswordWrappedInFormik = () => {
+  type Values = {
+    password: string
+  }
+
+  const TestForm = () => (
+    <Formik<Values>
+      initialValues={{
+        password: '',
+      }}
+      initialErrors={{
+        password: 'Server Error: Invalid password',
+      }}
+      validate={(values) => {
+        const errors: Partial<{ [key in keyof Values]: string }> = {
+          password: values.password.length < 4 ? 'Password is too short' : undefined,
+        }
+
+        return errors
+      }}
+      onSubmit={(values) => logger.log('submit', values)}
+    >
+      {({ values }) => (
+        <Form>
+          Values: {JSON.stringify(values)}
+          <PasswordFieldFormik<Values> name="password" label="Name" />
```

Comment:
Since the other PR is already merged, please let's add a submit button and field level validation for consistency.